src/testing/fast_array_utils/pytest.py

```diff
@@ -1,150 +1,69 @@
-) -> ToArray:
-    """Create a function to convert to a supported array."""
-    if array_cls is np.ndarray:
-        return np.asarray  # type: ignore[return-value]
-    if array_cls is types.DaskArray:
-        assert array_cls_name is not None
-        return to_dask_array(array_cls_name)
-    if array_cls is types.H5Dataset:
-        assert request is not None
-        return request.getfixturevalue("to_h5py_dataset")  # type: ignore[no-any-return]
-    if array_cls is types.ZarrArray:
-        return to_zarr_array
-    if array_cls is types.CupyArray:
-        import cupy as cu
-
-        return cu.asarray  # type: ignore[no-any-return]
-
-    return array_cls  # type: ignore[return-value]
-
-
-def _half_chunk_size(a: tuple[int, ...]) -> tuple[int, ...]:
-    def half_rounded_up(x: int) -> int:
-        div, mod = divmod(x, 2)
-        return div + (mod > 0)
-
-    return tuple(half_rounded_up(x) for x in a)
-
-
-def to_dask_array(array_cls_name: str) -> ToArray:
-    """Convert to a dask array."""
-    if TYPE_CHECKING:
-        import dask.array.core as da
-    else:
-        import dask.array as da
-
-    inner_cls_name = array_cls_name.removeprefix("dask.array.Array[").removesuffix("]")
-    inner_cls = get_array_cls(inner_cls_name)
-    to_array_fn: ToArray = get_to_array(array_cls=inner_cls)
-
-    def to_dask_array(x: ArrayLike, *, dtype: DTypeLike | None = None) -> types.DaskArray:
-        x = np.asarray(x, dtype=dtype)
-        return da.from_array(to_array_fn(x), _half_chunk_size(x.shape))  # type: ignore[no-untyped-call,no-any-return]
-
-    return to_dask_array
+    at = cast(ArrayType, request.param)
+    if at.cls is H5Dataset:
+        ctx = request.getfixturevalue("conversion_context")
+        at = dataclasses.replace(at, conversion_context=ctx)
+    return at
 
 
 @pytest.fixture(scope="session")
 # worker_id for xdist since we don't want to override open files
-def to_h5py_dataset(
+def conversion_context(
     tmp_path_factory: pytest.TempPathFactory,
     worker_id: str = "serial",
-) -> Generator[ToArray, None, None]:
-    """Convert to a h5py dataset."""
+) -> Generator[ConversionContext, None, None]:
+    """Fixture providing a :class:`~testing.fast_array_utils.ConversionContext`.
+
+    Makes sure h5py works even when running tests in parallel.
+    """
     import h5py
 
     tmp_path = tmp_path_factory.mktemp("backed_adata")
     tmp_path = tmp_path / f"test_{worker_id}.h5ad"
 
-    with h5py.File(tmp_path, "x") as f:
-
-        def to_h5py_dataset(x: ArrayLike, *, dtype: DTypeLike | None = None) -> types.H5Dataset:
-            arr = np.asarray(x, dtype=dtype)
-            test_name = os.environ["PYTEST_CURRENT_TEST"].rsplit(":", 1)[-1].split(" ", 1)[0]
-            return f.create_dataset(test_name, arr.shape, arr.dtype, data=arr)
-
-        yield to_h5py_dataset
+    def get_ds_name() -> str:
+        return os.environ["PYTEST_CURRENT_TEST"].rsplit(":", 1)[-1].split(" ", 1)[0]
```

Comment:
Could you leave a comment what this does and why it needs to be test-dependent? I think I understand, but am not 100% sure

Reply:
done